### DISCUSSION TITLE
Is Vike possbile with C# Asp.Net backend using cshtml for rendering?

### DISCUSSION THREAD

**Discussion by terwort** [vikejs/vike#1190] [Discussion]
Hello! I just wanted to ask a question about using this within my project. My current project is sort of a hybrid beast. Most of our stuff comes from C# ASP.NET razor views, and then another part of the app comes from hooking in our vite app after the initial view from the C# backend is loaded on the client. We create the initial react tree using a div that gets rendered from one of the cshtml files, and some portals to put various things where we need them. 

So the question I have is, would Vike be possible to use in this sort of environment? I still need to use the asp.net routing so I'd need to use a catch-all page route on the Vike app? Just trying to see if anyone could tell me off the top of their head if its even possible with my current technologies. 

I am trying to find a solution for SSR and possibly SSG for guest users, and this looks very promising. Thanks!

**Comment by brillout** [comment]
I ain't familiar with ASP.NET nor with your application architecture so I don't think it's possible for me to provide guidance without spending more time on your specific requirements. That said, in genereal, I'd recommend having a clear speration of concerns where all the frontend rendering is done solely by your Vike/React code, while having your backend logic handled by ASP.NET. For example, a catch-all route doesn't feel right (but, again, I don't know enough to say with certainty).

> looks very promising

👍 Let us know if you believe there is something about Vike that is limiting you.
